### PR TITLE
Implemented user dao

### PR DESCRIPTION
**Pull request by ethanliu24**
Implemented the DAO for users. Some codes are commented out for now to avoid the program not building, since the user entities are not modified/reviewed yet - once that's done those code can be uncommented and the DAO will be working correctly.

### CODE REVIEW

File: src/main/java/daos/DBUserDataAccessObject.java [hunk 1]
```diff
@@ -1,39 +1,158 @@
...
+        Bson projectionFields = Projections.fields(
+            Projections.include(field),
+            Projections.excludeId()
+        );
+        
+        Document doc = this.userRepository
+            .find(eq(field, target))
+            .projection(projectionFields)
+            .first();
+
```
Comment (Karththigan) [inline src/main/java/daos/DBUserDataAccessObject.java:129]:
```suggestion
        if (doc == null) {
            throw new UserNotFoundException("No user found with " + field + ": " + target);
        }
```

File: src/main/java/daos/DBUserDataAccessObject.java [hunk 2]
```diff
@@ -1,39 +1,158 @@
 package daos;
 
+import entity.CommonUser;
 import entity.User;
+import entity.UserFactory;
```
Comment (Karththigan) [inline src/main/java/daos/DBUserDataAccessObject.java]: I think I might have the same issue in my PR, but shouldn't the DAO layer be decoupled from entities to follow CA rules?

Should we have an intermediate layer that converts our User entities to MongoDB schemas?

**Review by Karththigan — APPROVED** [review]
Looks good for now! Just a few smaller comments, but feel free to merge

File: src/main/java/daos/DBUserDataAccessObject.java (hunk 2 above)
Comment (ethanliu24) [reply src/main/java/daos/DBUserDataAccessObject.java]: I've replied to ur comment in that PR, I'll check with the TA in lab on monday, make a change and merge it in

File: src/main/java/daos/DBUserDataAccessObject.java (hunk 1 above)
Comment (ethanliu24) [reply src/main/java/daos/DBUserDataAccessObject.java:129]: I think the use cases could deal with this, some times we may want to determine if a data exists in the database, but throwing an error during the query probably breaks it

**Pull request merged**